pkg/hsskey/hsskey.go

```diff
@@ -148,31 +150,36 @@ func GetAllHss(warnings io.Writer, verboseDangerous bool, eepromPattern string,
 	// Attempt to get HSS from IPMI.
 	hssList, err := getHssFromIpmi(warnings, verboseDangerous)
 	if err != nil || len(hssList) == 0 {
-		io.WriteString(warnings, fmt.Sprintf("Failed to get HSS key from IPMI: %v\n", err))
+		fmt.Fprintf(warnings, "Failed to get HSS key from IPMI: %v\n", err)
 	}
 
 	// Attempt to get HSS from EEPROM.
 	if eepromPattern != "" {
 		filePaths, err := getHssEepromPaths(baseSysfsPattern, eepromPattern)
 		if err != nil {
-			io.WriteString(warnings, fmt.Sprintf("Failed to find HSS EEPROM paths: %v\n", err))
+			fmt.Fprintf(warnings, "Failed to find HSS EEPROM paths: %v\n", err)
 		}
 		hssEeprom, err := GetHssFromFile(warnings, verboseDangerous, filePaths, hostSecretSeedCount)
 		if err == nil && len(hssEeprom) > 0 {
 			hssList = append(hssList, hssEeprom...)
 		} else {
-			io.WriteString(warnings, fmt.Sprintf("Failed to get HSS key from file: %v\n", err))
+			fmt.Fprintf(warnings, "Failed to get HSS key from file: %v\n", err)
 		}
 	}
 
+	// Attempt to get HSS from files.
 	if hssFiles != "" {
-		hssFileArr := strings.Split(hssFiles, ",")
-		hss, err := GetHssFromFile(warnings, verboseDangerous, hssFileArr, 0)
+		hssFileArr, err := evaluateHssPaths(hssFiles)
 		if err != nil {
-			io.WriteString(warnings, fmt.Sprintf("Error parsing file %s for HSS: %v\n", hss, err))
-		}
-		if len(hss) > 0 {
-			hssList = append(hssList, hss...)
+			fmt.Fprintf(warnings, "Failed to evaluate HSS file paths: %v\n", err)
+		} else {
+			hss, err := GetHssFromFile(warnings, verboseDangerous, hssFileArr, 0)
+			if err != nil {
+				fmt.Fprintf(warnings, "Error parsing file %s for HSS: %v\n", hss, err)
+			}
+			if len(hss) > 0 {
```

Comment:
the test here for len(hss) > 0 is not needed I think?just do the append?

Reply:
:) apologies, misread your initial request for this. Kind of neat finding that nil works as an empty slice. Removed the if.